modules/go-sequencer-client/client/client.go

```diff
@@ -24,27 +25,92 @@ type BalanceResponse struct {
 
 // Client is an HTTP tendermint client.
 type Client struct {
-	client *http.HTTP
+	websocket string
+	client    *http.HTTP
 }
 
 func NewClient(url string) (*Client, error) {
+	// Compile the regular expression
+	re := regexp.MustCompile(`^[^:]+://`)
+
 	c, err := http.New(url, "")
 	if err != nil {
 		return nil, err
 	}
 
+	// Replace and print results
+	websocket := re.ReplaceAllString(url, "")
+	websocket = "tcp://" + websocket
 	return &Client{
-		client: c,
+		websocket: websocket,
+		client:    c,
 	}, nil
 }
 
+// BroadcastTx broadcasts a transaction. If async is true, the function will
+// return immediately. The response seen is the generated data used for
+// submitting the transaction. It does not confirm that the data has been
+// included on chain. If async is false, the function will wait for the
+// transaction to be seen on the network.
+func (c *Client) BroadcastTx(ctx context.Context, tx *txproto.SignedTransaction, async bool) (*coretypes.ResultBroadcastTx, error) {
+	if async {
+		return c.BroadcastTxAsync(ctx, tx)
+	}
+	return c.BroadcastTxSync(ctx, tx)
+}
+
+// BroadcastTxAsync broadcasts a transaction and returns immediately.
+func (c *Client) BroadcastTxAsync(ctx context.Context, tx *txproto.SignedTransaction) (*coretypes.ResultBroadcastTx, error) {
+	bytes, err := proto.Marshal(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	return c.client.BroadcastTxAsync(ctx, bytes)
+}
+
+// BroadcastTxSync broadcasts a transaction and waits for the response that
+// confirms the transaction was included.
 func (c *Client) BroadcastTxSync(ctx context.Context, tx *txproto.SignedTransaction) (*coretypes.ResultBroadcastTx, error) {
 	bytes, err := proto.Marshal(tx)
 	if err != nil {
 		return nil, err
 	}
+	result, resultErr := c.client.BroadcastTxSync(ctx, bytes)
+
+	// Create a new RPC client
+	wsClient, err := http.New(c.websocket, "/websocket")
+	if err != nil {
+		log.Fatalf("Failed to create client: %v", err)
+	}
+	// close the client on function exit
+	defer func(wsClient *http.HTTP) {
+		err := wsClient.Stop()
+		if err != nil {
+			log.Fatalf("Failed to stop WebSocket client: %v", err)
+		}
+	}(wsClient)
+
+	// Start the WebSocket client
+	err = wsClient.Start()
+	if err != nil {
+		log.Fatalf("Failed to start WebSocket client: %v", err)
+	}
 
-	return c.client.BroadcastTxSync(ctx, bytes)
+	// Subscribe to transaction events
+	query := fmt.Sprintf("tm.event = 'Tx' AND tx.hash = '%X'", result.Hash)
+	out, err := wsClient.Subscribe(ctx, "clientID", query)
+	if err != nil {
+		log.Fatalf("Failed to subscribe to events: %v", err)
+	}
+
+	// Wait for and handle results from the channel
+	for event := range out {
```

Comment:
For posterity:
I don't think events are sent back for all txs, e.g. add fee asset doesn't seem to send an event even for a successful tx